imod/mf6/utilities/regrid.py

```diff
@@ -1,26 +1,29 @@
 import abc
-from enum import Enum
-from typing import Any, Optional, Tuple, Union
+import copy
```

Comment:
It's good that all the regrid code is now out of the packages, models and simulation. However the next step is to refactor this file and split it up in smaller files. At the moment its rather overwhelming. We can probably convert this in a separate module:


regrid/
├─ regridder_instances_collection.py
├─ model_regridder.py
├─ package_regridder.py
├─ RegridderType.py
├─ utilities.py
├─ __ init__.py
├─ ...

Notice that I also added the RegridderType That one can moved to the regrid module aswell.
It also makes it easy/transparent for the user what the can use fromt the module because we can explicitly add the methods to be used in __ init__.py file as well as documentation on how to use regridding

Reply:
sounds good, but that's not what the issue describes. @JoerivanEngelen , what about this?

Reply:
No you're right. It should be a follow up story. Keeping it as a separate commit also makes it easier to review